com.unity.ml-agents/Runtime/Policies/BehaviorParameters.cs

```diff
@@ -161,5 +160,15 @@ internal IPolicy GeneratePolicy(Func<float[]> heuristic)
                     return new HeuristicPolicy(heuristic);
             }
         }
+
+        internal void UpdateAgentPolicy()
+        {
+            var agent = GetComponent<Agent>();
```

Comment:
This might be an expensive call, but we can take care of this later...

Reply:
Maybe, but it only happens when the user does something, so it's at least avoidable in the general case.